### PR TITLE
UIEH-978: Add a Selection Status filter to Search within Packages modal for Title Record

### PR DESCRIPTION
**Pull request by Dmitriy-Litvinenko**
## Description
-  Add a Selection Status filter to Search within Packages modal for Title Record

## Issue
[UIEH-978](https://issues.folio.org/browse/UIEH-978)

## Screenshots
image

### CODE REVIEW

File: src/components/title/show/package-filter-modal/package-filter-modal.js [hunk 1]
```diff
@@ -50,13 +63,32 @@ const PackageFilterModal = ({
...
+      .filter(
+        ({ id: packageId }) => (
+          selectedOptions.length
+            ? selectedOptions.some((({ value: optionId }) => packageId === optionId))
+            : true
+        )
+      )
+      .filter(
+        ({ isSelected, providerId }) => (
+          searchFilters.selected
```
Comment (BogdanDenis) [inline src/components/title/show/package-filter-modal/package-filter-modal.js:81]: Please split this up so there's no complex ternary checks. It's a bit difficult to understand now

**Review by BogdanDenis — CHANGES_REQUESTED** [review]

File: src/components/title/show/package-filter-modal/package-filter-modal.js (hunk 1 above)
Comment (Dmitriy-Litvinenko) [reply src/components/title/show/package-filter-modal/package-filter-modal.js]: Fixed

File: src/components/title/show/package-filter-modal/package-filter-modal.js [hunk 2]
```diff
@@ -50,13 +63,34 @@ const PackageFilterModal = ({
...
+        ? isSelected && providerId === ebscoProviderId
+        : searchFilters.selected === isSelected.toString();
+    };
+
+    const newSelectedPackages = allPackages
+      .filter(
+        ({ id: packageId }) => (
+          selectedOptions.length
+            ? selectedOptions.some((({ value: optionId }) => packageId === optionId))
+            : true
```
Comment (Vladyslav-Velytskyi) [inline src/components/title/show/package-filter-modal/package-filter-modal.js:86]: Please, try this way
```suggestion
          !selectedOptions.length || selectedOptions.some((({ value: optionId }) => packageId === optionId))
```

File: src/components/title/show/package-filter-modal/package-filter-modal.js [hunk 3]
```diff
@@ -93,27 +129,50 @@ const PackageFilterModal = ({
...
+                ]
+              }}
+            />
+          </Accordion>
+          <SearchFilters
+            searchType={searchTypes.PACKAGES}
+            activeFilters={searchFilters}
+            availableFilters={[
+              selectionStatusFilterConfig,
+            ]}
```
Comment (Vladyslav-Velytskyi) [inline src/components/title/show/package-filter-modal/package-filter-modal.js:171]: Didn't see any reason why it should be written in separate strings
```suggestion
            availableFilters={[selectionStatusFilterConfig]}
```

File: src/constants/filterConfigs.js [hunk 4]
```diff
@@ -31,15 +31,38 @@ export const titleSortFilterConfig = {
...
 };
 
+export const selectionStatusFilterOptions = {
+  ALL: 'all',
+  TRUE: 'true',
+  FALSE: 'false',
+  EBSCO: 'ebsco',
+};
+
+export const ebscoProviderId = 19;
```
Comment (Vladyslav-Velytskyi) [inline src/constants/filterConfigs.js:41]: For constant variables for number, better to name them in uppercase
```suggestion
export const EBSCO_PROVIDER_ID = 19;
```

File: src/components/title/show/package-filter-modal/package-filter-modal.js (hunk 3 above)
Comment (Dmitriy-Litvinenko) [reply src/components/title/show/package-filter-modal/package-filter-modal.js:171]: Fixed

File: src/constants/filterConfigs.js (hunk 4 above)
Comment (Dmitriy-Litvinenko) [reply src/constants/filterConfigs.js:41]: Fixed

File: test/bigtest/tests/title-show-package-filter-test.js [hunk 5]
```diff
@@ -72,13 +72,84 @@ describe('TitleShow package filter flow', () => {
...
+      it('selection filter accordion should be opened by default', () => {
+        expect(PackageFilterModal.selectionFilterAccordion.isOpen).to.be.true;
+      });
+
+      it('should display the package filter selection filter', () => {
+        expect(PackageFilterModal.selectionFilterIsPresent).to.be.true;
+      });
+
+      describe('Selection filter', () => {
+        describe('Selection filter selection status all', () => {
```
Comment (Vladyslav-Velytskyi) [inline test/bigtest/tests/title-show-package-filter-test.js:96]: Better to call block of test with `when` word, for exp. 
```suggestion
        describe('when status "All" selected for selection filter', () => {
```

File: test/bigtest/tests/title-show-package-filter-test.js [hunk 6]
```diff
@@ -72,13 +72,84 @@ describe('TitleShow package filter flow', () => {
...
+
+          it('should select all from selection status', () => {
+            expect(PackageFilterModal.selectionFilterAll.isChecked).to.be.true;
+            expect(PackageFilterModal.selectionFilterSelected.isChecked).to.be.false;
+            expect(PackageFilterModal.selectionFilterNotSelected.isChecked).to.be.false;
+            expect(PackageFilterModal.selectionFilterEBSCO.isChecked).to.be.false;
+          });
+        });
+
+        describe('Selection filter selection status selected', () => {
```
Comment (Vladyslav-Velytskyi) [inline test/bigtest/tests/title-show-package-filter-test.js:109]:
```suggestion
        describe('when status "Selected" selected for selection filter', () => {
```

File: test/bigtest/tests/title-show-package-filter-test.js [hunk 7]
```diff
@@ -72,13 +72,84 @@ describe('TitleShow package filter flow', () => {
...
+
+          it('should select selected from selection status', () => {
+            expect(PackageFilterModal.selectionFilterAll.isChecked).to.be.false;
+            expect(PackageFilterModal.selectionFilterSelected.isChecked).to.be.true;
+            expect(PackageFilterModal.selectionFilterNotSelected.isChecked).to.be.false;
+            expect(PackageFilterModal.selectionFilterEBSCO.isChecked).to.be.false;
+          });
+        });
+
+        describe('Selection filter selection status not selected', () => {
```
Comment (Vladyslav-Velytskyi) [inline test/bigtest/tests/title-show-package-filter-test.js:122]:
```suggestion
        describe('when status "Not selected" for selection filter', () => {
```

File: test/bigtest/tests/title-show-package-filter-test.js [hunk 8]
```diff
@@ -111,6 +182,22 @@ describe('TitleShow package filter flow', () => {
...
             expect(this.location.search).to.equal(`?filteredPackages%5B0%5D=${title.resources.models[0].id}`);
           });
+
+          it('should display that one filter was selected', () => {
+            expect(PackageFilterModal.filterCount).to.equal('1');
+          });
+        });
+      });
+
+      describe('Selected filters', () => {
```
Comment (Vladyslav-Velytskyi) [inline test/bigtest/tests/title-show-package-filter-test.js:192]:
```suggestion
      describe('when selected more than one filters', () => {
```

**Review by Vladyslav-Velytskyi — COMMENTED** [review]
Looks good, please, update test names to make them more readable

File: test/bigtest/tests/title-show-package-filter-test.js (hunk 5 above)
Comment (Dmitriy-Litvinenko) [reply test/bigtest/tests/title-show-package-filter-test.js]: Fixed

File: test/bigtest/tests/title-show-package-filter-test.js (hunk 6 above)
Comment (Dmitriy-Litvinenko) [reply test/bigtest/tests/title-show-package-filter-test.js]: Fixed

File: test/bigtest/tests/title-show-package-filter-test.js (hunk 7 above)
Comment (Dmitriy-Litvinenko) [reply test/bigtest/tests/title-show-package-filter-test.js]: Fixed

File: test/bigtest/tests/title-show-package-filter-test.js (hunk 8 above)
Comment (Dmitriy-Litvinenko) [reply test/bigtest/tests/title-show-package-filter-test.js]: Fixed

**Review by Vladyslav-Velytskyi — APPROVED** [review]

**Review by BogdanDenis — APPROVED** [review]

**Pull request merged**